### DISCUSSION TITLE
tasmota.publish

### DISCUSSION THREAD

**Discussion by mtibi75** [arendst/Tasmota#21308] [Ideas]
tasmota.publish command "spamming" web console. It could be the same as tasmota.cmd mute parameter? Or other method to disable berry's generated mqtt publish log?

**Comment by s-hadinger** [comment]
Currently not possible, the internal MQTT publish function always logs at `LOG_LEVEL_INFO` level. I will check with Theo when he is back if we can add a flag to avoid logging on specific circumstances.

**Reply by ottelo9** [reply]
Is there any news/consideration to disable logging for MQTT?

**Comment by arendst** [comment]
I'll prepare function `void MqttPublishPayload(const char* topic, const char* payload, uint32_t binary_length, bool retained, uint32_t log_level = LOG_LEVEL_INFO)` to allow call level selection of log level which will be default LOG_LEVEL_INFO.

@s-hadinger then over to you to find a solution within berry to allow users to select LOG_LEVEL_NONE (or any other allowable level).

**Reply by arendst** [reply]
My part done https://github.com/arendst/Tasmota/commit/a932b3d991caf9911614dbb77bc6382c64b65c46

**Reply by ottelo9** [reply]
Thx for your answer. Sadly This Solution would disable all other maybe important messages. Is it not possible to add a switch to disable all MQTT messages / logs ?

**Reply by arendst** [reply]
No it doesn't. It all depends how @s-hadinger will implement it's use. I forsee a mute option for your tasmota.publish command which mutes only for this publish as you requested. All other MQTT messages will still be logged.

BTW I see no reason for this request to implement; the console is just there to log messages....

**Reply by s-hadinger** [reply]
Ok I will add a 'mute' argument, later this week.

**Reply by ottelo9** [reply]
> No it doesn't. It all depends how @s-hadinger will implement it's use. I forsee a mute option for your tasmota.publish command which mutes only for this publish as you requested. All other MQTT messages will still be logged.
> 
> BTW I see no reason for this request to implement; the console is just there to log messages....

I use Tasmota Scripting + SML. And if you use this meter config line (transmit the data immediately over MQTT), you will get an MQTT message every second in the console. This means I can't really use the console anymore, as all the other maybe useful messages get lost in the mix. Just 1 or 2 lines of that and the console gets completely spammed.

1,77070100100700ff@1,Leistung,W,Power_curr,**16**

**Reply by sfromis** [reply]
Doing something to publish messages every second is outside of how Tasmota is meant to be used.

Anyway, it should be doable to change the scripter to support doing publish with the new log level option, instead of a sledgehammer tactic like "suppress all".

**Reply by ottelo9** [reply]
Thx, @gemu2015 has implemented it inside SML driver.
